From 9db155965d0f8faba21241ad65b2718b2b58fddb Mon Sep 17 00:00:00 2001
From: Sveinung Kvilhaugsvik <sveinung84@users.sourceforge.net>
Date: Sat, 30 Jan 2016 18:23:34 +0100
Subject: [PATCH] JSON: specify delta vector field address.

This makes the combination of the delta protocol and JSON work again.

Reported by Marko Lindqvist <cazfi>

See bug #24354
---
 common/generate_packets.py | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/common/generate_packets.py b/common/generate_packets.py
index 21ca068..9f3ffb3 100755
--- a/common/generate_packets.py
+++ b/common/generate_packets.py
@@ -1120,6 +1120,9 @@ static char *stats_%(name)s_names[] = {%(names)s};
 '''%self.get_dict(vars())
 
         body=body+'''
+  #ifdef FREECIV_JSON_CONNECTION
+  field_addr.name = "fields";
+  #endif /* FREECIV_JSON_CONNECTION */
   DIO_BV_PUT(&dout, \"fields\", &field_addr, fields);
 '''
 
@@ -1161,7 +1164,12 @@ static char *stats_%(name)s_names[] = {%(names)s};
   struct %(packet_name)s *old;
   struct genhash **hash = pc->phs.received + %(type)s;
 '''
-            delta_body1="\n  DIO_BV_GET(&din, \"fields\", &field_addr, fields);\n"
+            delta_body1='''
+  #ifdef FREECIV_JSON_CONNECTION
+  field_addr.name = "fields";
+  #endif /* FREECIV_JSON_CONNECTION */
+  DIO_BV_GET(&din, \"fields\", &field_addr, fields);
+  '''
             body1=""
             for field in self.key_fields:
                 body1=body1+prefix("  ",field.get_get())+"\n"
-- 
2.7.0.rc3

